### PR TITLE
Add information about moving storage locations and improve lifecycle guide

### PR DESCRIPTION
**Pull request by hsanjuan**

### CODE REVIEW

**Review by johnnymatthews — CHANGES_REQUESTED** [review]

**Comment by hsanjuan** [comment]
let's wait on @lanzafame for final review on this

File: docs/mine/lotus/miner-lifecycle.md
Comment (mishmosh) [inline docs/mine/lotus/miner-lifecycle.md:17]:
```suggestion
Before shutting down your miner daemon, make sure that there are no pending operations that could get your miner _slashed_ if they are delayed or cancelled.
```

Comment (mishmosh) [inline docs/mine/lotus/miner-lifecycle.md:17]: It would be nice to have some expectation-setting like "A simple restart usually takes about X minutes" and any other caveats around time.

**Review by mishmosh — APPROVED** [review]

File: docs/mine/lotus/miner-lifecycle.md
Comment (lanzafame) [reply docs/mine/lotus/miner-lifecycle.md:17]: A restart will take as long as it takes for your machine to verify the proof params

File: docs/mine/lotus/custom-storage-layout.md
Comment (lanzafame) [inline docs/mine/lotus/custom-storage-layout.md:13]:
```suggestion
The Lotus Miner keeps track of defined storage locations in in `~/.lotusminer/storage.json` (or `$LOTUS_MINER_PATH/storage.json`) and uses `~/.lotusminer` path as default.
```

File: docs/mine/lotus/miner-lifecycle.md
Comment (lanzafame) [reply docs/mine/lotus/miner-lifecycle.md:17]: This sentence should point to a section on how to check the mpool for pending messages.

Comment (lanzafame) [reply docs/mine/lotus/miner-lifecycle.md:17]: As such it would be a good suggestion that miners put the params onto nvme drives so that they can be quickly read.

File: docs/mine/lotus/miner-lifecycle.md
Comment (lanzafame) [inline docs/mine/lotus/miner-lifecycle.md:72]:
```suggestion
::: warning
There is a maximum of three attempts to complete the _precommit2_ operation before sealing is fully started from scratch (_pre-commit1_ phase).
```

File: docs/mine/lotus/miner-lifecycle.md
Comment (lanzafame) [inline docs/mine/lotus/miner-lifecycle.md:83]:
```suggestion
The above command will give you an overview of [storage locations known to the miner](custom-storage-layout.md). This information is stored in `~/.lotusminer/storage.json` (or `$LOTUS_MINER_PATH/storage.json` if defined). Lotus seal workers store all the data in the `~/.lotusworker` folder (or `$LOTUS_WORKER_PATH` if defined).
```

File: docs/mine/lotus/miner-lifecycle.md
Comment (lanzafame) [inline docs/mine/lotus/miner-lifecycle.md:88]:
```suggestion
1. Copy or move the data. You should move the data folder from the original location to the destination without changing any of the content.
2. [Stop the miner](#safely-restarting-the-miner-daemon)
```
Stopping the miner first will result in definitely missing many window posts purely due to the amount of time it will take to transfer any non-trivial number of sectors. i.e. 1GbE LAN takes just over ~1hr to transfer 448GiBs (or 14 sectors).

File: docs/mine/lotus/miner-lifecycle.md
Comment (lanzafame) [inline docs/mine/lotus/miner-lifecycle.md:120]: I don't think lotus currently supports offline importing of keys, but I could be mistaken.

File: docs/mine/lotus/seal-workers.md
Comment (lanzafame) [inline docs/mine/lotus/seal-workers.md:23]:
```suggestion
The Lotus Worker stores all the necessary data in the `~/.lotusworker` folder. You can customize that by setting the `$LOTUS_WORKER_PATH` environment variable. It is recommended to put this folder on a NVMe SSD so that it gets faster disk performance.
```

**Review by lanzafame — CHANGES_REQUESTED** [review]

File: docs/mine/lotus/miner-lifecycle.md
Comment (hsanjuan) [reply docs/mine/lotus/miner-lifecycle.md:17]: Isn't enough with the `proving info` below (or `proving dearlines`)?

File: docs/mine/lotus/miner-lifecycle.md
Comment (hsanjuan) [reply docs/mine/lotus/miner-lifecycle.md:88]: I'm guessing the data should not be moved then, while the miner is running, only copied. And I am guessing that even if the miner is running, it should not take on any new deals etc while the copy is ongoing?

File: docs/mine/lotus/miner-lifecycle.md
Comment (hsanjuan) [reply docs/mine/lotus/miner-lifecycle.md:120]: This assumes both old and new node are running. Export/import instructions also mention just copying the `keystore` folder though, I'll make it clearer.

File: docs/mine/lotus/seal-workers.md
Comment (hsanjuan) [reply docs/mine/lotus/seal-workers.md:23]: I really screwed all the paths. I will grep for any wrong ones.

File: docs/mine/lotus/miner-lifecycle.md
Comment (hsanjuan) [reply docs/mine/lotus/miner-lifecycle.md:17]: > how to check the mpool for pending messages.

Can you actually tell me how?

Comment (hsanjuan) [reply docs/mine/lotus/miner-lifecycle.md:17]: Another question is how to make a miner stop taking deals without restarting it (so that we can be sure it has not just closed another deal and it's receiving data when its stopped).

Comment (hsanjuan) [reply docs/mine/lotus/miner-lifecycle.md:17]: added this suggestion to the setup instructions.

**Comment by mishmosh** [comment]
Tagging @ribasushi for review as well since he's been very hands-on answering miner questions on this topic.

File: docs/mine/lotus/miner-lifecycle.md
Comment (lanzafame) [reply docs/mine/lotus/miner-lifecycle.md:17]: Actually, yes to restart just the miner checking proving is sufficient. But in case here is how you check pending messages, `lotus mpool pending --local | jq -c . | wc -l` will give you the number of pending messages. To stop taking deals you can do `lotus-miner storage-deals selection reject --online --offline` and `lotus-miner retrieval-deals selection reject --online --offline`

File: docs/mine/lotus/miner-lifecycle.md
Comment (lanzafame) [reply docs/mine/lotus/miner-lifecycle.md:88]: > I'm guessing the data should not be moved then, while the miner is running, only copied

correct
> it should not take on any new deals etc while the copy is ongoing?

correct, also don't start pledging any new sectors, obviously.

**Review by lanzafame — APPROVED** [review]

**Comment by hsanjuan** [comment]
I will be making a follow up PR with improvements to the "restarting your miner" section, which has received many comments but it is not originally the scope of this change, so I don't want to block it on that discussion.

**Pull request merged**
